src/libclipper/include/clipper/task_executor.hpp

```diff
@@ -272,6 +287,16 @@ class TaskExecutor {
             VersionedModelId vm = VersionedModelId(
                 container_info["model_name"], container_info["model_version"]);
             int replica_id = std::stoi(container_info["model_replica_id"]);
+            //#int batch_size = std::stoi(container_info["batch_size"]);
+            boost::optional<int> batch_size = -1;
+            for( auto it = container_info.begin(); it != container_info.end(); it++){
```

Comment:
Instead of iterating over the key-value pairs in the `container_info` map, use the [find](http://en.cppreference.com/w/cpp/container/unordered_map/find) function and check the validity result of the "find" operation as:

```
auto &batch_size_search = container_info.find("batch_size")
if(batch_size_search != container_info.end()) {
    // The entry exists, obtain the value and update the batch size
    ...
}
```